tests/drivers/pwm/pwm_gpio_loopback/testcase.yaml

```diff
@@ -0,0 +1,11 @@
+tests:
+  drivers.pwm.gpio_loopback:
+    tags:
+      - drivers
+      - pwm
+    filter: (dt_alias_exists("pwm-0") or dt_alias_exists("pwm-1") or dt_alias_exists("pwm-2")
+      or dt_alias_exists("pwm-3")) and dt_compat_enabled("test-pwm-gpio-loopback")
```

Comment:
Replace this filter with soc specific test cases, and use platform allow to select which boards should be tested. This ties in with the snippets mentioned in another comment, see https://github.com/zephyrproject-rtos/zephyr/blob/d789f339efb78a136a10c74efdeeb963e085e104/tests/drivers/comparator/gpio_loopback/testcase.yaml#L14-L20